libminifi/src/core/yaml/YamlConfiguration.cpp

```diff
@@ -339,7 +339,7 @@ void YamlConfiguration::parseProvenanceReportingYaml(
     return;
   }
 
-  if (!reportNode || !(reportNode->IsSequence())) {
```

Comment:
The main case of interest is for those scenarios were Provenance Reporting is not desired.  As a result, this will throw an exception for those scenarios where provenance reporting is not one of the included yaml components.